### PR TITLE
Fix #926 windeployqt and other cmake broken stuff

### LINKED ISSUE #926: Qt Windows install is broken
As well as windeplyqt which is not called.
Good news I have a fix
Bad new, to be effective on all plateform, whithout adding "if plateform", we need to update to cmake 3.19
Any feedback ?

### PR DESCRIPTION
**Pull request by dlyr**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix qt windows compilation

* **What is the current behavior?** (You can also link to an open issue here)
fixup_blundle do not find qt dll's and windeployqt is not called

* **What is the new behavior (if this is a feature change)?**
Fix theses ;)

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
Yes, because cmake get_target_properties is used with new behavior introduced in cmake 3.19, client need to have relatively recent cmake, which is not included in debian bullseye for instance (need to use backport).

* **Other information**:
TODO
- [x] update doc
- [x] update cmake version in all necessary files
- [x] test @MathiasPaulin mention about multiple inclusion
- [x] fix this dummy warning /W3 /W4 which is still present ...
- [x] clean history

### CODE REVIEW

File: cmake/RadiumSetupFunctions.cmake [hunk 1]
```diff
@@ -6,7 +6,7 @@
 # Minimum version: 3.13: install target from different directory see
```
Comment (MathiasPaulin) [inline cmake/RadiumSetupFunctions.cmake:6]: perhaps you can state here why minimum version is 3.19 (instead of 3.13)

File: src/Config.cmake.in [hunk 2]
```diff
@@ -27,8 +27,11 @@ endif("${isSystemDir}" STREQUAL "-1")
 
 # --------------------------------------------------------------------------------------------------
 # Configure components
-@PACKAGE_INIT@
 
+# Load Radium cmake functions
+include(${CMAKE_CURRENT_LIST_DIR}/RadiumSetupFunctions.cmake)
```
Comment (MathiasPaulin) [inline src/Config.cmake.in:32]: This needs to be verified here. What's happen if this was already included ? I remember I had such a case when searching for the Radium package from a subdirectory of the buildtree (e.g making find package(Radium ...) in a CMakeFile.txt in subdir of thee radium source tree) or when several components are searching for Radium. I will not be available to test this next week. I'll do that later if the PR is still opened.

**Review by MathiasPaulin — COMMENTED** [review]
There is also a problem when calling windeploy Qt in Debug mode. As the debug option is not passed to windeployQt, the Qt dll deployed are not the good ones. I fixed this with N....as B....so. And I think this fix should also be integrated in this PR.

**Comment by dlyr** [comment]
Tested in debug mode, no problems, install radium and install an external project which use radium both work on windows.

**Comment by dlyr** [comment]
Actually, I have made a quick test (under linux) cmake 3.19 is not needed, so we can rollback a bit, but I don't know how to find correct minimal cmake version.
I can check one or two cmake version under linux to see which is sufficient.
At least 3.18 is ok.

**Comment by dlyr** [comment]
Just tested under linux with cmake 3.16, it works, but cmake 3.6 failed (do not support position independent -C option).
It works also with cmake 3.10 on linux ... do someone want to test under windows ?

**Review by MathiasPaulin — APPROVED** [review]

**Pull request merged**